cli/cli_app.go

```diff
@@ -878,81 +884,92 @@ func remoteStateNeedsInit(remoteState *remote.RemoteState, terragruntOptions *op
+// on terraform_remote_state dependencies.
+func applyAll(terragruntOptions *options.TerragruntOptions) error {
+	opts := terragruntOptions.Clone(terragruntOptions.TerragruntConfigPath)
+	opts.TerraformCommand = "apply"
+	opts.OriginalTerraformCommand = "apply"
+	opts.TerraformCliArgs = append([]string{"apply"}, opts.TerraformCliArgs...)
+	logCommandAllDeprecationWarning(opts, "apply-all")
+	return runAll(opts)
 }
 
 // Tear down an entire "stack" by running 'terragrunt destroy' in each subfolder, processing them in the right order
 // based on terraform_remote_state dependencies.
 func destroyAll(terragruntOptions *options.TerragruntOptions) error {
-	stack, err := configstack.FindStackInSubfolders(terragruntOptions)
-	if err != nil {
-		return err
-	}
-
-	terragruntOptions.Logger.Printf("%s", stack.String())
-	shouldDestroyAll, err := shell.PromptUserForYesNo("WARNING: Are you sure you want to run `terragrunt destroy` in each folder of the stack described above? There is no undo!", terragruntOptions)
-	if err != nil {
-		return err
-	}
-
-	if shouldDestroyAll {
-		return stack.Destroy(terragruntOptions)
-	}
-
-	return nil
+	opts := terragruntOptions.Clone(terragruntOptions.TerragruntConfigPath)
+	opts.TerraformCommand = "destroy"
+	opts.OriginalTerraformCommand = "destroy"
+	opts.TerraformCliArgs = append([]string{"destroy"}, opts.TerraformCliArgs...)
+	logCommandAllDeprecationWarning(opts, "destroy-all")
+	return runAll(opts)
 }
 
 // outputAll prints the outputs from all configuration in a stack, in the order
 // specified in the terraform_remote_state dependencies
 func outputAll(terragruntOptions *options.TerragruntOptions) error {
-	stack, err := configstack.FindStackInSubfolders(terragruntOptions)
-	if err != nil {
-		return err
-	}
-
-	terragruntOptions.Logger.Printf("%s", stack.String())
-	return stack.Output(terragruntOptions)
+	opts := terragruntOptions.Clone(terragruntOptions.TerragruntConfigPath)
+	opts.TerraformCommand = "output"
+	opts.OriginalTerraformCommand = "output"
+	opts.TerraformCliArgs = append([]string{"output"}, opts.TerraformCliArgs...)
+	logCommandAllDeprecationWarning(opts, "output-all")
+	return runAll(opts)
 }
 
 // validateAll validates runs terraform validate on all the modules
 func validateAll(terragruntOptions *options.TerragruntOptions) error {
-	stack, err := configstack.FindStackInSubfolders(terragruntOptions)
-	if err != nil {
-		return err
-	}
+	opts := terragruntOptions.Clone(terragruntOptions.TerragruntConfigPath)
+	opts.TerraformCommand = "validate"
+	opts.OriginalTerraformCommand = "validate"
+	opts.TerraformCliArgs = append([]string{"validate"}, opts.TerraformCliArgs...)
+	logCommandAllDeprecationWarning(opts, "validate-all")
+	return runAll(opts)
+}
 
-	terragruntOptions.Logger.Printf("%s", stack.String())
-	return stack.Validate(terragruntOptions)
+func logCommandAllDeprecationWarning(terragruntOptions *options.TerragruntOptions, command string) {
```

Comment:
See also: https://github.com/gruntwork-io/terragrunt/pull/1510/files#diff-43e9ba6ef3ee5016f1ec4acc37037441ca46ffcbf0221417035e48c4393bfbc9R313... We'll want to go with one of these prob.

Reply:
Good catch! Consolidated into a more functional deprecation handler: 8fb85adcdf9fcc1061b393903448e71b8f1732aa